-1

Tested at: 6152acab52d0a2a9c47de7aae88558b7f5dbddaf

CMSSW: CMSSW_11_1_X_2020-03-16-1100
SCRAM_ARCH: slc7_amd64_gcc820
You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-cdb816/5199/summary.html

I found follow errors while testing this PR

**Failed tests:  Build  HeaderConsistency  ClangBuild**

* **Build**:

I found compilation error when building: 

<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/EventFilter/CTPPSRawToDigi/src/RawDataUnpacker.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/EventFilter/CTPPSRawToDigi/src/VFATFrame.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/EventFilter/CTPPSRawToDigi/src/CTPPSPixelDataFormatter.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/EventFilter/CTPPSRawToDigi/src/RPixErrorChecker.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/EventFilter/CTPPSRawToDigi/src/RPixErrorChecker.cc: In member function 'bool RPixErrorChecker::checkROC(bool&, int, uint32_t, const Word32&, RPixErrorChecker::Errors&) const':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/EventFilter/CTPPSRawToDigi/src/RPixErrorChecker.cc:76:5: error: expected '(' before 'LIKELY'
     LIKELY(errorType < 25) return true;
     ^~~~~~
     (
gmake: *** [tmp/slc7_amd64_gcc820/src/EventFilter/CTPPSRawToDigi/src/EventFilterCTPPSRawToDigi/RPixErrorChecker.cc.o] Error 1
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/EventFilter/CTPPSRawToDigi/src/CTPPSPixelDataFormatter.cc: In member function 'void CTPPSPixelDataFormatter::interpretRawData(bool&, int, const FEDRawData&, CTPPSPixelDataFormatter::Collection&, CTPPSPixelDataFormatter::Errors&)':
</pre>

* **Clang**:

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/CondFormats/SiPixelObjects/src/SiPixelGainCalibration.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/CondFormats/SiPixelObjects/src/SiPixelFedCablingMap.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/CondFormats/SiPixelObjects/src/T_EventSetup_SiPixelFedCablingMap.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/CondFormats/SiPixelObjects/src/T_EventSetup_SiPixelCalibConfiguration.cc 
>> Compiling  /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/CondFormats/SiPixelObjects/src/SiPixelCPEGenericErrorParm.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2020-03-16-1100/src/CondFormats/SiPixelObjects/src/SiPixelFrameConverter.cc:29:5: error: expected '(' after 'if'
    UNLIKELY(!rocp) {
    ^
1 error generated.
gmake: *** [tmp/slc7_amd64_gcc820/src/CondFormats/SiPixelObjects/src/CondFormatsSiPixelObjects/SiPixelFrameConverter.cc.o] Error 1
>> Compiling  tmp/slc7_amd64_gcc820/src/CondFormats/SiPixelObjects/src/CondFormatsSiPixelObjects/a/Serialization.cc 
</pre>

--- a/DataFormats/Common/interface/BasicHandle.h
+++ b/DataFormats/Common/interface/BasicHandle.h
@@ -28,7 +28,6 @@ If failedToGet() returns false but isValid() is also false then no attempt
 #include "DataFormats/Provenance/interface/ProductID.h"
 #include "DataFormats/Provenance/interface/Provenance.h"
 #include "DataFormats/Common/interface/HandleExceptionFactory.h"
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 
 #include <memory>
 

--- a/DataFormats/Common/interface/DetSet.h
+++ b/DataFormats/Common/interface/DetSet.h
@@ -15,7 +15,6 @@ own copy of the common DetId.
 #include <cstdint>
 #include <cstddef>
 #include "DataFormats/Common/interface/CMS_CLASS_VERSION.h"
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 
 namespace edm {
   typedef uint32_t det_id_type;

--- a/DataFormats/Common/interface/Holder.h
+++ b/DataFormats/Common/interface/Holder.h
@@ -4,7 +4,6 @@
 #include "DataFormats/Common/interface/CMS_CLASS_VERSION.h"
 #include "DataFormats/Common/interface/BaseHolder.h"
 #include "DataFormats/Common/interface/RefHolder.h"
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 #include <memory>
 
 namespace edm {

--- a/DataFormats/Common/interface/MultiAssociation.h
+++ b/DataFormats/Common/interface/MultiAssociation.h
@@ -55,7 +55,6 @@
 #include <boost/range.hpp>
 #include "DataFormats/Common/interface/CMS_CLASS_VERSION.h"
 #include "DataFormats/Provenance/interface/ProductID.h"
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 
 namespace edm {
   namespace helper {

--- a/DataFormats/Common/interface/RefHolder_.h
+++ b/DataFormats/Common/interface/RefHolder_.h
@@ -5,7 +5,6 @@
 #include "DataFormats/Common/interface/RefHolderBase.h"
 #include "DataFormats/Provenance/interface/ProductID.h"
 #include "FWCore/Utilities/interface/OffsetToBase.h"
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 #include <memory>
 #include <typeinfo>
 

--- a/DataFormats/Common/interface/SortedCollection.h
+++ b/DataFormats/Common/interface/SortedCollection.h
@@ -33,8 +33,6 @@ unreliable if such duplicate entries are made.
 #include "DataFormats/Provenance/interface/ProductID.h"
 #include "FWCore/Utilities/interface/EDMException.h"
 
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
-
 #include <algorithm>
 #include <typeinfo>
 #include <vector>

--- a/DataFormats/Common/interface/VectorHolder.h
+++ b/DataFormats/Common/interface/VectorHolder.h
@@ -4,7 +4,6 @@
 #include "DataFormats/Common/interface/BaseVectorHolder.h"
 #include "DataFormats/Common/interface/Holder.h"
 #include <memory>
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 
 namespace edm {
   namespace reftobase {

--- a/DataFormats/Common/interface/Wrapper.h
+++ b/DataFormats/Common/interface/Wrapper.h
@@ -11,7 +11,6 @@ Wrapper: A template wrapper around EDProducts to hold the product ID.
 #include "DataFormats/Common/interface/WrapperDetail.h"
 #include "DataFormats/Common/interface/CMS_CLASS_VERSION.h"
 #include "DataFormats/Provenance/interface/ProductID.h"
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 #include "FWCore/Utilities/interface/Visibility.h"
 
 #include <algorithm>

--- a/FWCore/Framework/test/stubs/TestBeginEndJobAnalyzer.h
+++ b/FWCore/Framework/test/stubs/TestBeginEndJobAnalyzer.h
@@ -20,7 +20,6 @@
 
 // system include files
 #include "FWCore/Framework/interface/EDAnalyzer.h"
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 // user include files
 
 // forward declarations

--- a/FWCore/Utilities/interface/Exception.h
+++ b/FWCore/Utilities/interface/Exception.h
@@ -38,8 +38,9 @@
 #include <exception>
 #include <type_traits>
 
-#include "FWCore/Utilities/interface/GCC11Compatibility.h"
 #include "FWCore/Utilities/interface/thread_safety_macros.h"
+#include "FWCore/Utilities/interface/Likely.h"
+#include "FWCore/Utilities/interface/Visibility.h"
 
 namespace cms {
 

--- a/FWCore/Utilities/interface/GCC11Compatibility.h
+++ b/FWCore/Utilities/interface/GCC11Compatibility.h
@@ -1,42 +1,4 @@
 #ifndef FWCORE_GCC11COMPATIBILITY_H
 #define FWCORE_GCC11COMPATIBILITY_H
-/*
- * set of macro to control the level of comaptibility with c++11 standard
- * includes also other macro more specific to gcc
-
- */
-
-#include "FWCore/Utilities/interface/Visibility.h"
-#include "FWCore/Utilities/interface/Likely.h"
-
-#if defined(__GCCXML__) || defined(__CINT__)
-#define CMS_NOCXX11
-#endif
-#if !GCC_PREREQUISITE(4, 6, 0) && !defined(__clang__)
-#define CMS_NOCXX11
-#endif
-
-#ifndef CMS_NOCXX11
-#if GCC_PREREQUISITE(4, 7, 0)
-#define GCC11_FINAL final
-#define GCC11_OVERRIDE override
-#elif __clang__
-// FIXME: do not know how to query for final support. It's there in any
-//        clang version we will ever use, so it's fine.
-#define GCC11_FINAL final
-#if __has_feature(cxx_override_control)
-#define GCC11_OVERRIDE override
-#endif
-#else
-#define GCC11_FINAL
-#define GCC11_OVERRIDE
-#endif  // gcc 4.7
-#else
-#define constexpr
-#define noexcept
-#define nullptr 0
-#define GCC11_FINAL
-#define GCC11_OVERRIDE
-#endif  // NOCXX11
-
+#warning "GCC11Compatibility.h should be replaced with Visibility.h or Likely.h"
 #endif  // FWCORE_GCC11COMPATIBILITY_H

